pkg/standalone-utils/contracts/test/MockTetuSmartVault.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity >=0.7.0 <0.9.0;
+
+import "@balancer-labs/v2-interfaces/contracts/standalone-utils/ITetuSmartVault.sol";
+
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/SafeERC20.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/test/TestToken.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/math/FixedPoint.sol";
+
+import "./MockTetuStrategy.sol";
+import "../relayer/special/TetuShareValueHelper.sol";
+
+contract MockTetuSmartVault is ITetuSmartVault, TestToken, TetuShareValueHelper {
+    using SafeERC20 for IERC20;
+    using FixedPoint for uint256;
+
+    IERC20 public underlyingAsset;
+    uint256 private immutable _underlyingDecimals;
+    uint256 private _underlyingBalanceInVault = 0;
+    MockTetuStrategy private immutable _tetuStrategy;
+    uint256 _desiredRate;
+
+    constructor(
+        string memory name,
+        string memory symbol,
+        uint8 decimals,
+        address _underlyingAsset,
+        MockTetuStrategy tetuStrategy
+    ) TestToken(name, symbol, decimals) {
+        underlyingAsset = IERC20(_underlyingAsset);
+        _underlyingDecimals = decimals;
+        _tetuStrategy = tetuStrategy;
+    }
+
+    function getPricePerFullShare() external pure override returns (uint256) {
+        revert("Should not call this");
+    }
+
+    // Should pass rate with decimals from underlyingToken
+    function setRate(uint256 newRate) public {
+        // stores latest rate, so the balances are recalculated after deposit and withdraw
+        _desiredRate = newRate;
```

Comment:
Maybe I'm just picky, but it bothers me that this function sets `_desiredRate`, and we also sometimes call `setRate(_desiredRate)`, so that it overwrites itself.

It would feel nicer if we had something like a private `_setRate()` that we could call without setting `_desiredRate`. Then the public `setRate()` would be reduced to just a couple lines: first set `_desiredRate`, then call `_setRate()`.